src/Symfony/Component/Form/Extension/Core/Type/MultiStepType.php

```diff
@@ -0,0 +1,174 @@
+    {
+        $resolver
+            ->setRequired('steps')
+            ->setAllowedTypes('steps', 'array')
+            ->setAllowedValues('steps', static function (array $steps): bool {
+                foreach ($steps as $key => $step) {
+                    if (!\is_string($key)) {
+                        return false;
+                    }
+
+                    if ((!\is_string($step) || !is_subclass_of($step, AbstractType::class)) && !\is_callable($step)) {
+                        return false;
+                    }
+                }
+
+                return true;
+            })
+            ->setRequired('current_step')
+            ->setAllowedTypes('current_step', 'string')
+            ->setNormalizer('current_step', static function (Options $options, string $value): string {
+                if (!\array_key_exists($value, $options['steps'])) {
+                    throw new InvalidOptionsException(\sprintf('The current step "%s" does not exist.', $value));
+                }
+
+                return $value;
+            })
+            ->setDefault('current_step', static function (Options $options): string {
+                /** @var string $firstStep */
+                $firstStep = array_key_first($options['steps']);
+
+                return $firstStep;
+            })
+            ->setRequired('next_step')
+            ->setAllowedTypes('next_step', ['string', 'null'])
+            ->setDefault('next_step', function (Options $options): ?string {
+                return array_keys($options['steps'])[$this->currentStepIndex($options['current_step'], $options['steps']) + 1] ?? null;
+            })
+            ->setNormalizer('next_step', static function (Options $options, ?string $value): ?string {
+                if (null === $value) {
+                    return null;
+                }
+
+                if (!\array_key_exists($value, $options['steps'])) {
+                    throw new InvalidOptionsException(\sprintf('The next step "%s" does not exist.', $value));
+                }
+
+                return $value;
+            })
+            ->setRequired('previous_step')
+            ->setAllowedTypes('previous_step', ['string', 'null'])
+            ->setDefault('previous_step', function (Options $options): ?string {
+                return array_keys($options['steps'])[$this->currentStepIndex($options['current_step'], $options['steps']) - 1] ?? null;
+            })
+            ->setNormalizer('previous_step', static function (Options $options, ?string $value): ?string {
+                if (null === $value) {
+                    return null;
+                }
+
+                if (!\array_key_exists($value, $options['steps'])) {
+                    throw new InvalidOptionsException(\sprintf('The previous step "%s" does not exist.', $value));
+                }
+
+                return $value;
+            });
+
+        $resolver->setDefaults([
+            'hide_back_button_on_first_step' => false,
```

Comment:
display_ with default true instead of hide_ default false?
Seems easier to read the positive way not the négative